src/Security/Authorization/Core/src/AuthorizationOptions.cs

```diff
@@ -27,6 +27,14 @@ public class AuthorizationOptions
         /// </remarks>
         public AuthorizationPolicy DefaultPolicy { get; set; } = new AuthorizationPolicyBuilder().RequireAuthenticatedUser().Build();
 
+        /// <summary>
+        /// Gets or sets the fallback authorization policy used by <see cref="AuthorizationPolicy.CombineAsync(IAuthorizationPolicyProvider, IEnumerable{IAuthorizeData})"/>
+        /// when no IAuthorizeData have been provided. As a result, the <see cref="AuthorizationMiddleware"/> uses the fallback policy
+        /// if there are no <see cref="IAuthorizeData"/> instances for a resource. If a resource has any <see cref="IAuthorizeData"/>
+        /// then they are evaluated instead of the fallback policy. By default the fallback policy is null, and is not used in any way by the default <see cref="IAuthorizationService"/>.
+        /// </summary>
+        public AuthorizationPolicy FallbackPolicy { get; set; }
```

Comment:
Should we add a note that says this is unsupported by MVC and SignalR when not using endpoint routing?

Reply:
yes you should, otherwise I'll forget what this is again, and ask for it to be removed ;)

Reply:
Is that true?  I thought the middleware would always apply fallback policy even if there's no endpoint metadata, this logic comes from Combine...

When there's no endpoint routing, that just means metadata = null, which should still use fallback policy (just there would be no resource)?

Reply:
We do not enforce that the authZ middleware is configured when you do not have endpoint metadata. Your app will function with no indication that the fallback policy is being ignored. While you could add the authz middleware and have it work, I think it's much better to say that for SignalR and MVC, the recommended way is to use endpoint routing for this feature to work reliably.

Reply:
So is your concern, that they might not have UseAuthorization() at all?  In that case, they don't have the middleware so nothing would be using the fallback policy, and that seems ok since the comments do specifically say this is something the middleware does.

My question was more whether there was any situation where the fallback policy would be ignored WHEN they do have UseAuthorization() in their pipeline?

Reply:
Would it be clearer if I just add an explicit sentence at saying something like "fallbackPolicy usually will have no effect unless you have the AuthZ middleware in your pipeline."

Reply:
> fallbackPolicy usually will have no effect unless you have the AuthZ middleware in your pipeline."

That sounds great.